src/build.sh

```diff
@@ -1,21 +1,55 @@
 #!/usr/bin/env bash
 set -eu
 
-die() {
-    echo "*** ERROR; build failed"
-    exit 1
-}
+#===============================================================================
+# Assuming this script is run with: `cd $GERBIL_BASE/src && ./build.sh`
+# All other build scripts depend on this point of entry
+#===============================================================================
+
+## global (for all build scripts) setup
+readonly GERBIL_SOURCE=$(pwd -P)
+readonly GERBIL_BASE="$(dirname "${GERBIL_SOURCE}")"
+readonly BUILD_DIR="${GERBIL_SOURCE}/build"
+readonly GERBIL_SETUP="" # this flag signals that basic setup has occured
```

Comment:
occurred

Reply:
I did this because bash is using tri-state logic: unset, null(empty), else. The variable being set to any value (either empty or 1 or any other value) means that some script touched it (since the default value is unset). My reasoning was that if some script bothered to set a value, then, presumably, it also handled the setup.

Looking back, it would have been more readable to ignore the tri-state logic and just test for 1.

I guess its a moot point since all that control logic is now gone :D